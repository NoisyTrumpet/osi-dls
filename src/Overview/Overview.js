import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './Overview.scss';
import { Typography } from '../components/Typography';
import { version, homepage } from '../../package.json';

// eslint-disable-next-line import/prefer-default-export
export const Overview = ({ addClass, backgroundColor }) => {
  const introductionClasses =
    addClass && addClass.split(' ').map((i) => `introduction-${i}`);
  const backgroundColorClass = {
    [`background-color-${backgroundColor}`]: backgroundColor,
  };
  const classes = classNames(
    'introduction',
    introductionClasses,
    backgroundColorClass,
  );
  return (
    <div className={classes}>
      <Typography variant="headline-6">Introduction</Typography>
      <Typography variant="subtext">
        The Design Language System (DLS) contains several React components such
        as buttons, form fields, etc which can be imported in React and are
        self-contained with styling, test suites, functional JSX code, etc. For
        detailed readme, issues, suggestions, and related projects, view the{' '}
        <a href={homepage}>
          {' '}
          Project Source
        </a>{' '}
        on GitHub.
      </Typography>
      <br />
      <Typography variant="body-medium">{`DLS v${version}`}</Typography>
    </div>
  );
};

Overview.defaultProps = {};

Overview.propTypes = {
  addClass: PropTypes.string,
  backgroundColor: PropTypes.string,
};
