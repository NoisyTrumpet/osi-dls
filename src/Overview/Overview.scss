.introduction {
  display: flex;
  justify-content: center;
  flex-direction: column;
  height: 100vh;
  margin: auto;
  max-width: 30em;


  a {
    color: rgba(map-get($colors, 'primary'), 1.0);

    &:hover {
      text-decoration: none;
    }
  }
}
