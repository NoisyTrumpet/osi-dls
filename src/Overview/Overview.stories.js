import React from "react";
import { storiesOf } from "@storybook/react";
import { Overview } from "./Overview";

storiesOf("overview | Introduction", module).add("Introduction", () => (
  <Overview />
));
