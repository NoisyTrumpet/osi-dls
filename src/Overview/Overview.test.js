import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Overview } from './Overview';

describe('<Overview />', () => {
  test('Should render', () => {
    const overview = shallow(<Overview />);
    expect(overview.length).toBe(1);
    expect(toJson(overview)).toMatchSnapshot();
  });
});
