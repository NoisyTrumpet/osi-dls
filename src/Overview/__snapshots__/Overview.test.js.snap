// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overview /> Should render 1`] = `
<div
  className="introduction"
>
  <Typography
    variant="headline-6"
  >
    Introduction
  </Typography>
  <Typography
    variant="subtext"
  >
    The Design Language System (DLS) contains several React components such as buttons, form fields, etc which can be imported in React and are self-contained with styling, test suites, functional JSX code, etc. For detailed readme, issues, suggestions, and related projects, view the
     
    <a
      href="https://github.com/NoisyTrumpet/osi-dls#readme"
    >
       
      Project Source
    </a>
     
    on GitHub.
  </Typography>
  <br />
  <Typography
    variant="body-medium"
  >
    DLS v1.5.0
  </Typography>
</div>
`;
